### DISCUSSION TITLE
What about Requests&Responses folders move to Shared project?

### DISCUSSION THREAD

**Discussion by biproberkay** [fullstackhero/blazor-starter-kit#387] [Ideas]
I was looking for 👀 : req&res classes. 
realised : they all used by both client and server projects. 
tought 🤔: what if move theese classes to the shared?
doubted first : dependensies maybe not reliable for this idea 
controlled the dependensies : there was no except data annotations. 
finally i agreed with that idea,
my  suggestion is moving them to Shared project may be useful...💡
what do you thing ?

**Comment by Zigimun** [comment]
This is the correct approch... Since now there is a bug that the Client is dependant on the Application project which exposes the code to the client.

**Reply by biproberkay** [reply]
![image](https://user-images.githubusercontent.com/58739068/147877763-e644c492-7f6e-41e2-96ee-a0662604902d.png)
https://github.com/blazorhero/CleanArchitecture/blob/a4d049847f6875fe46ee9b9598167846ac0df9a4/src/Client/_Imports.razor#L42
here it is. reference is only for requests.

**Comment by jeremiedevos** [comment]
In my opinion, it might indeed be better to move it to a shared library.
Maybe in some use cases it might be even better to just add a separate library that you can make available for download, so others can use the same models when using the API. However, that might be too specific to a situation.

**Comment by biproberkay** [comment]
actually there is no concern of seperation back-end and front end codes here. when there was a concern, this req res classes must be both in their own repositories. so ı think front end ui layers web>client>client.shared must be in a diffirent repository to use api on other spa frameworks mobile or other platforms.
